chirp/drivers/tdh8.py

```diff
@@ -1243,77 +1201,19 @@ def _get_scan_list(self, scan_data):
 
         return ([x_list, y_list])
 
-    def get_block_xy(self, number):
-        if number % 8 != 0:
-            x_list = number / 8
-            y_list = number % 8
-
-        else:
-            x_list = (number / 8) - 1
-            y_list = 8
-
-        return ([x_list, y_list])
-
     def set_memory(self, mem):
         _mem = self._get_mem(mem.number)
         _nam = self._get_nam(mem.number)
 
+        # When the channel is empty, you need to set "usedflags" to 0,
+        # When the channel is used , you need to set "usedflags" to 1.
+        self._memobj.usedflags[mem.number - 1] = int(~mem.empty)
```

Comment:
Noice, this is very clean. But, I think you should use `not` here instead of `~`:
```
>>> not False
True
>>> ~False
-1
```
So you're actually setting -1 instead of 1, which based on the bit math in the bitwise object will make this weird. I assume you tested this with the radio?

```
>>> 0x80 |  (-1 << 3)
-8
>>> 0x80 |  (1 << 3)
136
```
I think it will still be okay because of the masking that goes on, but it's not super great for understandability in the future :)